### PR TITLE
[FIX][9.0] NL BTW statement, onchange date range

### LINKED ISSUE #87: NL BTW statement, date range not working
When creating a new NL BTW Statement I select a date range. Unfortunately the date range is not used in de two date fields in the NL BTW Statement form.

### PR DESCRIPTION
**Pull request by astirpe**
Fixes #87

### CODE REVIEW

**Comment by astirpe** [comment]
Runbot failure not related with this PR:

```
2017-06-21 07:40:07,624 174 ERROR openerp_test openerp.modules.graph: module l10n_nl_postcodeapi: Unmet dependencies: partner_street_number

```

It is because the porting of `partner_street_number` module is still pending. See PR: OCA/partner-contact#207

**Review by hbrunn — APPROVED** [review]

**Comment by astirpe** [comment]
I'm also including the backport of the fix https://github.com/OCA/l10n-netherlands/pull/97

**Review by CasVissers-360ERP — APPROVED** [review]

**Comment by astirpe** [comment]
This PR got 2 approvals long time ago. The reason of the runbot failure is described here https://github.com/OCA/l10n-netherlands/pull/89#issuecomment-309994329. Every new PR on branch 9.0 will fail for that reason.

So in my opinion this PR can be merged.

**Comment by hbrunn** [comment]
we'll get an inconsistent repo this way. If you want to speed this up, I'd suggest taking over the partner_street_number branch, fix the last comments and increase test coverage. Then we can merge there and afterwards here

**Comment by astirpe** [comment]
The repo (branch 9.0) is already inconsistent. The cause is not related with this PR. The cause is related with the module l10n_nl_postcodeapi, that was merged with broken dependencies long time ago. I don't want to speed up this PR, but I don't see a valid reason to block an obvious back-porting of the bug fixing that was already accepted and merged in branch 10.

File: l10n_nl_tax_statement/views/l10n_nl_vat_statement.xml [hunk 1]
```diff
@@ -21,7 +21,7 @@
                     <group>
                         <group name="tax_report">
                             <field name="company_id" options="{'no_create': True}" groups="base.group_multi_company"/>
-                            <field name="date_range_id" domain="[('company_id','=',company_id)]" attrs="{'readonly': [('state','=','posted')]}"/>
```
Comment (StefanRijnhart) [inline l10n_nl_tax_statement/views/l10n_nl_vat_statement.xml:24]: I believe company_id is not required on date ranges, so you might want to allow for ranges without a company here.

File: l10n_nl_tax_statement/views/l10n_nl_vat_statement.xml [hunk 2]
```diff
@@ -21,7 +21,7 @@
...
+                            <field name="date_range_id" attrs="{'readonly': [('state','=','posted')]}"/>
```
Comment (astirpe) [inline l10n_nl_tax_statement/views/l10n_nl_vat_statement.xml:24]: This line fixes https://github.com/OCA/l10n-netherlands/issues/96 for V9

File: l10n_nl_tax_statement/views/l10n_nl_vat_statement.xml (hunk 1 above)
Comment (astirpe) [reply l10n_nl_tax_statement/views/l10n_nl_vat_statement.xml:24]: Yes, true. That's why I'm removing the domain in this commit 576473fdffae884d6902ffafc4cf788975a96f5c

Comment (StefanRijnhart) [reply l10n_nl_tax_statement/views/l10n_nl_vat_statement.xml:24]: LOL, sorry!

**Review by StefanRijnhart — APPROVED** [review]

**Comment by StefanRijnhart** [comment]
Agree with @astirpe that this does not break anything that is not already broken, so could be merged IMHO.

**Comment by hbrunn** [comment]
agreed, here i overdid it with my rule fetishism (which I consider a good thing in general). Sorry @astirpe and @StefanRijnhart, thanks for making this clear

**Pull request merged**